app/models/cluster_group.rb

```diff
@@ -315,6 +316,18 @@ def subsample_thresholds_required
     ClusterGroup::SUBSAMPLE_THRESHOLDS.select {|sample| sample < self.points}
   end
 
+  # control gate for invoking subsampling
+  def can_subsample?
+    if self.points < 1000 || self.subsampled
```

Comment:
Setting `MIN_THRESHOLD = 1000` and using that both here and in `SUBSAMPLE_THRESHOLDS` would avoid potential bugs caused by updating `1000` here but not in `SUBSAMPLE_THRESHOLDS` (or vice versa).

Reply:
How about `SUBSAMPLE_THRESHOLDS.min` instead?  Then it's only one place that needs updating, which I think is the goal of your suggestion.

Reply:
Sounds good!